direnvrc

```diff
@@ -1,5 +1,52 @@
 # shellcheck shell=bash
 
+_nix_import_env() {
+  local env=$1
+
+  local term_backup=$TERM path_backup=$PATH
+  if [[ -n ${TMPDIR+x} ]]; then
+    local tmp_backup=$TMPDIR
+  fi
+  local impure_ssl_cert_file=${SSL_CERT_FILE:-__UNSET__}
+  local impure_nix_ssl_cert_file=${NIX_SSL_CERT_FILE:-__UNSET__}
+
+  eval "$env"
+
+  # `nix-shell --pure` sets invalid ssl certificate paths
+  if [[ "${NIX_SSL_CERT_FILE:-}" = /no-cert-file.crt ]]; then
+    if [[ $1 == __UNSET__ ]]; then
+      unset NIX_SSL_CERT_FILE
+    else
+      export NIX_SSL_CERT_FILE=${impure_ssl_cert_file}
+    fi
+  fi
+
+  if [[ "${SSL_CERT_FILE:-}" = /no-cert-file.crt ]]; then
+    if [[ $impure_ssl_cert_file == __UNSET__ ]]; then
+      unset SSL_CERT_FILE
+    else
+      export SSL_CERT_FILE=$impure_nix_ssl_cert_file
+    fi
+  fi
+
+  export PATH=$PATH:$path_backup TERM=$term_backup
+  if [[ -n ${tmp_backup+x} ]]; then
+    export TMPDIR=${tmp_backup}
+  else
+    unset TMPDIR
+  fi
+
+  # misleading since we are in an impure shell now
+  export IN_NIX_SHELL=impure
+}
+
+use_flake() {
+  watch_file flake.nix
+  watch_file flake.lock
+
+  _nix_import_env "$(nix print-dev-env --profile "$(direnv_layout_dir)/flake-profile")"
```

Comment:
@zimbatm what do you think about this here?

```suggestion
  _nix_import_env "$(nix --no-net print-dev-env --profile "$(direnv_layout_dir)/flake-profile")"
```

I often encountered nix to forget about downloaded flakes. An alternative would be gc roots for all inputs.

Reply:
`--no-net` is new to me. I did a quick try on one of my repos and it started re-compiling perl and gettext. Without `--no-net` it was able to fetch those results from the cache.

I think the issue you are mentioning is legit but needs to be solved differently. The `--profile` only captures the build outputs, and the inputs can get garbage-collected. To solve that, one way would be to somehow capture all the inputs into the build output. Or create a separate buildEnv profile with the inputs.

Reply:
Actually I did a test and it also did not protect the outputs from beeing garbage collected.
I added a gcroot for the profile.

Reply:
I think the the directory specified in `--profile` must point to `/nix/var/nix/gcroots/per-user/$USER` to be effective.